code/datums/components/_component.dm

```diff
@@ -128,22 +128,24 @@
 /datum/component/proc/_RemoveFromParent()
 	var/datum/parent = src.parent
 	var/list/parents_components = parent._datum_components
-	for(var/I in _GetInverseTypeList())
-		var/list/components_of_type = parents_components[I]
 
-		if(length(components_of_type)) //
-			var/list/subtracted = components_of_type - src
+	if(parents_components)
+		for(var/I in _GetInverseTypeList())
+			var/list/components_of_type = parents_components[I]
 
-			if(subtracted.len == 1) //only 1 guy left
-				parents_components[I] = subtracted[1] //make him special
-			else
-				parents_components[I] = subtracted
+			if(length(components_of_type)) //
+				var/list/subtracted = components_of_type - src
+
+				if(subtracted.len == 1) //only 1 guy left
+					parents_components[I] = subtracted[1] //make him special
+				else
+					parents_components[I] = subtracted
 
-		else //just us
-			parents_components -= I
+			else //just us
+				parents_components -= I
 
-	if(!parents_components.len)
-		parent._datum_components = null
+		if(!parents_components.len)
+			parent._datum_components = null
```

Comment:
Could you revert these changes? It's because the hivebox is qdeling in initialize without passing the correct hints, I'm putting up a fix

Reply:
Applied with https://github.com/tgstation/tgstation/pull/77188/commits/26ec41da8eb659dbd5d6a295f7656e898adbc563